### ISSUE TITLE
No option to select spring-social integration

### ISSUE THREAD

**Issue by arabbani** [jhipster/generator-jhipster#7906]
##### **Overview of the issue**

When using with jhipster v5.0.1, there is no way to select spring-social feature during application generation

##### **Motivation for or Use Case**

Previous versions of jhipster had this feature

##### **Reproduce the error**

jhipster --jhi-prefix app

? Which *type* of application would you like to create? Monolithic application (recommended for simple projects)
? What is the base name of your application? FbQuiz
? What is your default Java package name? com.creatives.apsstr
? Do you want to use the JHipster Registry to configure, monitor and scale your application? No
? Which *type* of authentication would you like to use? HTTP Session Authentication (stateful, default Spring Security mechanism)
? Which *type* of database would you like to use? SQL (H2, MySQL, MariaDB, PostgreSQL, Oracle, MSSQL)
? Which *production* database would you like to use? MySQL
? Which *development* database would you like to use? MySQL
? Do you want to use the Spring cache abstraction? No - Warning, when using an SQL database, this will disable the Hibernate 2nd level cache!
? Would you like to use Maven or Gradle for building the backend? Maven
? Which other technologies would you like to use?
? Which *Framework* would you like to use for the client? Angular 6
? Would you like to enable *SASS* support using the LibSass stylesheet preprocessor? Yes
? Would you like to enable internationalization support? No
? Besides JUnit and Jest, which testing frameworks would you like to use?
? Would you like to install other generators from the JHipster Marketplace? No

##### **JHipster Version(s)**

JHipster v5.0.1

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.creatives.apsstr"
    },
    "jhipsterVersion": "5.0.1",
    "applicationType": "monolith",
    "baseName": "FbQuiz",
    "packageName": "com.creatives.apsstr",
    "packageFolder": "com/creatives/apsstr",
    "serverPort": "8080",
    "authenticationType": "session",
    "cacheProvider": "no",
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "mysql",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "rememberMeKey": "replaced-by-jhipster-info",
    "clientFramework": "angularX",
    "useSass": true,
    "clientPackageManager": "yarn",
    "testFrameworks": [],
    "jhiPrefix": "apsstr",
    "enableTranslation": false
  }
}
</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>

</pre>
</details>

##### **Environment and Tools**

java version "1.8.0_144"
Java(TM) SE Runtime Environment (build 1.8.0_144-b01)
Java HotSpot(TM) 64-Bit Server VM (build 25.144-b01, mixed mode)

git version 2.14.1.windows.1

node: v8.11.3

npm: 6.1.0

yarn: 1.7.0

##### **Browsers and Operating System**

Windows 10
Google Chrome

- [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by deepu105** [comment]
This option has been removed. Please see our release notes

**Comment by arabbani** [comment]
Any workaround?

**Comment by deepu105** [comment]
You could try this
https://www.jhipster.tech/tips/024_tip_enabling_social_login_with_oauth2.html

Thanks & Regards,
Deepu

**Comment by deepu105** [comment]
@jdubois I think we missed to mention this on the release notes for 5.0 or was it another one?

**Comment by jdubois** [comment]
Is should have been in a beta release, at least that was openly discussed many times, including several times on Twitter. 
I don't get it, nobody was interested in this feature and nobody wanted to contribute, but now it's removed we have many complaining it was removed. 
Anyway, the real solution is to have an OpenId Connect configuration that works for Keycloak/Okta/Google/Github/JHipster UAA but that's the same problem we have nobody available to contribute on this part.

**Comment by deepu105** [comment]
@jdubois I checked the release notes of Beta, we missed it. I guess we need to mention this somewhere in the docs

**Comment by arabbani** [comment]
Thanks a lot

**Comment by arabbani** [comment]
i missed the discussion. It was a very useful feature

**Comment by jdubois** [comment]
Yes, well the idea is that it should comes back with the new Spring Security OpenID Connect implementation that is under development, but we'll need help for this, we clearly don't have the manpower to do everything at the moment.
BTW there is another solution as of today: you can use Keycloak or Okta, and connect them with Google.

**Comment by PierreBesson** [comment]
If I remember well, the previous social login feature was only available for session auth which was a problem as it was not compatible with jhipster microservices, so it's logical it was phased out.

**Comment by jkili** [comment]
Just stumbled across this issue after spending a lot of time trying to work out where all the spring social code had gone after an upgrade from 4.14.4. Can I suggest you add it to the release notes for 5.0 so that the next person doesn't get caught out like I did?

Now I am pulling all the old spring social code and fitting it manually.

**Comment by jdubois** [comment]
Sorry @jkili !! Normally the documentation is updated everywhere, but we might have missed stuff, the website is so big now... Indeed I'm not sure it's in the release notes, I'll check this tomorrow.
I hope we can get it back at some point: it should work well with our current OAuth2 code, and hopefully it will be even better with the upcoming Spring Security 5. It's just supposed to be another OAuth2 provider, basically. So if you have something working, can you update us? Maybe even do a PR?

**Comment by jdubois** [comment]
Documentation updated at https://github.com/jhipster/jhipster.github.io/commit/aa9b7f0e7dc72442b8cdda6b3d562b04797b76f4 so I'm closing this.

@jkili if you have time to test with our current OAuth2 code, please send us some feedback - you can even create a specific PR for that.

**Comment by amrnablus** [comment]
Hi, is this a part of the beta 6 release? I can test it if that's the case

**Comment by jdubois** [comment]
Sorry but Spring Social isn't supported anymore since JHipster 5 - the Spring Social 2.0 release is now more than 2 years late, and I don't think they work on it anymore.

**Comment by amrnablus** [comment]
Ah i see, Thanks man

**Comment by unknowedNp** [comment]
@jdubois 
yeah , spring-social disconted but can you custom implemented or oauth.
https://spring.io/blog/2018/03/06/using-spring-security-5-to-integrate-with-oauth-2-secured-services-such-as-facebook-and-github

may have option to use in jwt or session security use social login without spring security ?

**Comment by jdubois** [comment]
Yes those should work either using our normal OAuth2 support, or through an OIDC provider like Keycloak or Okta

**Comment by amrnablus** [comment]
I switched to keycloak + oauth and it does exactly what spring social did. Anyone should be good switching over to that.
